include/mbedtls/ssl.h

```diff
@@ -629,11 +631,7 @@ union mbedtls_ssl_premaster_secret
 
 #define MBEDTLS_PREMASTER_SIZE     sizeof( union mbedtls_ssl_premaster_secret )
 
-#if defined(MBEDTLS_USE_PSA_CRYPTO)
-#define MBEDTLS_TLS1_3_MD_MAX_SIZE         PSA_HASH_MAX_SIZE
-#else
-#define MBEDTLS_TLS1_3_MD_MAX_SIZE         MBEDTLS_MD_MAX_SIZE
-#endif /* MBEDTLS_USE_PSA_CRYPTO */
+#define MBEDTLS_TLS1_3_MD_MAX_SIZE         MBEDTLS_HASH_MAX_SIZE
```

Comment:
Shouldn't this be `PSA_HASH_MAX_SIZE`? I could see that `psa_hash` is used everywhere in TLS1.3, not `mbedtls_md`.

Reply:
This should be `PSA_HASH_MAX_SIZE` only if this macro is never used to size buffers for code shared with TLS 1.2 (which uses the legacy API). Considering the name of the macro this seems unlikely, but I'll double-check uses before making the changes you suggest.

Reply:
Just a remark that may help here. This macro was introduced because TLS 1.3 ciphersuites only allow SHA-256 and SHA-384 not SHA-512. While SHA-512 may be included in the library and even used for some TLS 1.3 hash calculations where the hash is not the hash of the selected ciphersuite (not completely sure if there is such a case though, maybe for certificate chain validation ...), SHA-512 is not used fo hash calculations where the hash is the hash of the selected ciphersuite. Thus, this macro defines the maximum possible size of the hash of a selected ciphersuite. It is supposed to be equal eventually to MIN( PSA_HASH_MAX_SIZE, 384 ). Not the case yet and not documented because we never finalized that.

Reply:
Thanks for the info, @ronald-cron-arm ! I'll change to `PSA_HASH_MAX_SIZE` for now and let you take it from there.